Timing Mod wrapper rewrite

jedbakerMO opened the pull request:
# PR Summary

Sci/Tech Reviewer: @christophermaynard 
Code Reviewer: @mo-rickywong 

<!-- To be completed by the developer -->

<!-- Provide a brief description of the changes in this PR, including any notes
     useful for reviewers -->
This PR is the github migration of [Trac : lfric_core#4669](https://code.metoffice.gov.uk/trac/lfric/ticket/4669)
It replaces all timer calls throughout LFRic Apps (in the linked PR) and LFRic Core with the new timing wrapper. The timing wrapper can use either the current timer or Vernier, depending on suite configuration.
<!-- List any linked PRs here
- linked MetOffice/<REPO-NAME>#<pr-number>
-->
 linked MetOffice/lfric_apps#80

<!-- List any blocking PRs or issues to be closed here
- is blocked-by #pr-number
- blocks #pr-number
- closes #issue-number (auto-closes the issue)
- fixes #issue-number (auto-closes the issue)
- is related to #issue-number
-->
- closes #192 
- is related to  MetOffice/lfric_apps#68
- blocks Issue #196  which currently has no PR.
## Code Quality Checklist

(_Some checks are automatically carried out via the CI pipeline_)

- [X] I have performed a self-review of my own code
- [X] My code follows the project's
      [style guidelines](https://metoffice.github.io/lfric_core/how_to_contribute/index.html#how-to-contribute-index)
- [X] Comments have been included that aid understanding and enhance the
      readability of the code
- [X] My changes generate no new warnings

## Testing

- [X] I have tested this change locally, using the LFRic Core rose-stem suite
- [X] If required (e.g. API changes) I have also run the LFRic Apps test suite
      using this branch
- [ ] If any tests fail (rose-stem or CI) the reason is understood and
      acceptable (e.g. kgo changes)
- [ ] I have added tests to cover new functionality as appropriate (e.g. system
      tests, unit tests, etc.)
- [ ] Any new tests have been assigned an appropriate amount of compute resource
      and have been allocated to an appropriate testing group (i.e. the
      developer tests are for jobs which use a small amount of compute resource
      and complete in a matter of minutes)

<!-- Describe other testing performed (if applicable) -->

### trac.log

# Test Suite Results - lfric_core - timing_mod_core_updated/run1

## Suite Information

| Item | Value |
| :--- | :--- |
| Suite Name | timing_mod_core_updated/run1 |
| Suite User | jed.baker |
| Workflow Start | 2025-12-29T14:36:31 |
| Groups Run | developer |

| Dependency | Reference | Main Like |
| :--- | :--- | :--- |
| lfric_core | [jedbakerMO/lfric_core@192_timing_mod_updated_core](https://github.com/jedbakerMO/lfric_core/tree/192_timing_mod_updated_core) | False |
| SimSys_Scripts | [MetOffice/SimSys_Scripts@2025.12.1](https://github.com/MetOffice/SimSys_Scripts/tree/2025.12.1) | True |

## Task Information
<details>
<summary>:white_check_mark: succeeded tasks - 370</summary>

| Task | State |
| :--- | :--- |
| build_coupled_azspice_gnu_fast-debug-64bit | succeeded |
| build_coupled_azspice_gnu_full-debug-64bit | succeeded |
| build_coupled_ex1a_cce_fast-debug-64bit | succeeded |
| build_coupled_ex1a_cce_full-debug-64bit | succeeded |
| build_coupling_unit_tests_azspice_gnu_32bit | succeeded |
| build_coupling_unit_tests_azspice_gnu_64bit | succeeded |
| build_coupling_unit_tests_ex1a_gnu_32bit | succeeded |
| build_coupling_unit_tests_ex1a_gnu_64bit | succeeded |
| build_driver_unit_tests_azspice_gnu_32bit | succeeded |
| build_driver_unit_tests_azspice_gnu_64bit | succeeded |
| build_driver_unit_tests_ex1a_gnu_32bit | succeeded |
| build_driver_unit_tests_ex1a_gnu_64bit | succeeded |
| build_infrastructure_integration_tests_azspice_gnu_32bit | succeeded |
| build_infrastructure_integration_tests_azspice_gnu_64bit | succeeded |
| build_infrastructure_integration_tests_ex1a_cce_32bit | succeeded |
| build_infrastructure_integration_tests_ex1a_cce_64bit | succeeded |
| build_infrastructure_unit_tests_azspice_gnu_32bit | succeeded |
| build_infrastructure_unit_tests_azspice_gnu_64bit | succeeded |
| build_infrastructure_unit_tests_ex1a_gnu_32bit | succeeded |
| build_infrastructure_unit_tests_ex1a_gnu_64bit | succeeded |
| build_io_demo_azspice_gnu_fast-debug-32bit | succeeded |
| build_io_demo_azspice_gnu_fast-debug-64bit | succeeded |
| build_io_demo_azspice_gnu_full-debug-64bit | succeeded |
| build_io_demo_ex1a_cce_fast-debug-32bit | succeeded |
| build_io_demo_ex1a_cce_fast-debug-64bit | succeeded |
| build_io_demo_ex1a_cce_full-debug-64bit | succeeded |
| build_io_demo_ex1a_gnu_fast-debug-32bit | succeeded |
| build_io_demo_ex1a_gnu_fast-debug-64bit | succeeded |
| build_io_demo_unit_tests_azspice_gnu_64bit | succeeded |
| build_io_demo_unit_tests_ex1a_gnu_64bit | succeeded |
| build_lbc_demo_azspice_gnu_fast-debug-64bit | succeeded |
| build_lbc_demo_azspice_gnu_full-debug-64bit | succeeded |
| build_lbc_demo_ex1a_cce_fast-debug-64bit | succeeded |
| build_lbc_demo_ex1a_gnu_fast-debug-64bit | succeeded |
| build_lfric_xios_integration_tests_azspice_gnu_64bit | succeeded |
| build_lfric_xios_integration_tests_ex1a_cce_64bit | succeeded |
| build_lfric_xios_unit_tests_azspice_gnu_64bit | succeeded |
| build_lfric_xios_unit_tests_ex1a_gnu_64bit | succeeded |
| build_mesh_azspice_gnu_fast-debug-64bit | succeeded |
| build_mesh_ex1a_gnu_fast-debug-64bit | succeeded |
| build_mesh_tools_azspice_gnu_fast-debug-64bit | succeeded |
| build_mesh_tools_azspice_gnu_full-debug-64bit | succeeded |
| build_mesh_tools_ex1a_cce_fast-debug-64bit | succeeded |
| build_mesh_tools_ex1a_cce_full-debug-64bit | succeeded |
| build_mesh_tools_ex1a_gnu_fast-debug-64bit | succeeded |
| build_mesh_tools_unit_tests_azspice_gnu_64bit | succeeded |
| build_mesh_tools_unit_tests_ex1a_gnu_64bit | succeeded |
| build_science_unit_tests_azspice_gnu_32bit | succeeded |
| build_science_unit_tests_azspice_gnu_64bit | succeeded |
| build_science_unit_tests_ex1a_gnu_32bit | succeeded |
| build_science_unit_tests_ex1a_gnu_64bit | succeeded |
| build_simple_diffusion_azspice_gnu_fast-debug-64bit | succeeded |
| build_simple_diffusion_azspice_gnu_full-debug-64bit | succeeded |
| build_simple_diffusion_ex1a_cce_fast-debug-64bit | succeeded |
| build_simple_diffusion_ex1a_cce_full-debug-64bit | succeeded |
| build_simple_diffusion_ex1a_gnu_full-debug-64bit | succeeded |
| build_simple_diffusion_unit_tests_azspice_gnu_64bit | succeeded |
| build_simple_diffusion_unit_tests_ex1a_gnu_64bit | succeeded |
| build_skeleton_azspice_gnu_fast-debug-64bit | succeeded |
| build_skeleton_azspice_gnu_full-debug-64bit | succeeded |
| build_skeleton_ex1a_cce_full-debug-64bit | succeeded |
| build_skeleton_ex1a_gnu_fast-debug-64bit | succeeded |
| build_skeleton_ex1a_gnu_full-debug-64bit | succeeded |
| build_skeleton_unit_tests_azspice_gnu_64bit | succeeded |
| build_skeleton_unit_tests_ex1a_gnu_64bit | succeeded |
| check_coupled_default-C12_azspice_gnu_fast-debug-64bit | succeeded |
| check_coupled_default-C12_azspice_gnu_full-debug-64bit | succeeded |
| check_coupled_default-C12_ex1a_cce_fast-debug-64bit | succeeded |
| check_coupled_default-C12_ex1a_cce_full-debug-64bit | succeeded |
| check_io_demo_default-C24_azspice_gnu_fast-debug-32bit | succeeded |
| check_io_demo_default-C24_azspice_gnu_full-debug-64bit | succeeded |
| check_io_demo_default-C24_ex1a_cce_fast-debug-32bit | succeeded |
| check_io_demo_default-C24_ex1a_cce_full-debug-64bit | succeeded |
| check_io_demo_multifile-C24_azspice_gnu_fast-debug-32bit | succeeded |
| check_io_demo_multifile-C24_azspice_gnu_fast-debug-64bit | succeeded |
| check_io_demo_multifile-C24_ex1a_cce_fast-debug-32bit | succeeded |
| check_io_demo_multifile-C24_ex1a_cce_fast-debug-64bit | succeeded |
| check_io_demo_multifile-C24_ex1a_gnu_fast-debug-32bit | succeeded |
| check_io_demo_multifile-C24_ex1a_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_ConstantLBC-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_ConstantLBC-lbc_azspice_gnu_full-debug-64bit | succeeded |
| check_lbc_demo_ConstantLBC-lbc_ex1a_cce_fast-debug-64bit | succeeded |
| check_lbc_demo_ConstantLBC-lbc_ex1a_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_1x1P_ex1a_cce_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_1x1P_ex1a_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_2x2P_ex1a_cce_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_2x2P_ex1a_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_8x2P_ex1a_cce_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_8x2P_ex1a_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_OutputOnLBC-lbc_azspice_gnu_full-debug-64bit | succeeded |
| check_lbc_demo_default-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| check_lbc_demo_default-lbc_azspice_gnu_full-debug-64bit | succeeded |
| check_lbc_demo_default-lbc_ex1a_cce_fast-debug-64bit | succeeded |
| check_lbc_demo_default-lbc_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c1_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c1_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c1_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c2_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c2_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c2_ex1a_cce_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c3_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c3_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-c3_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-maps_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-maps_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-maps_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-op-nonuniform_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-op-nonuniform_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-op-nonuniform_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-op_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-op_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-op_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-rotated_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-rotated_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere-rotated_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_cubedsphere_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_equator-band_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_equator-band_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_equator-band_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_equator_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_equator_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_equator_ex1a_cce_full-debug-64bit | succeeded |
| check_mesh_tools_falklands_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_falklands_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_falklands_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_lam_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_lam_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_lam_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_london-model_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_london-model_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_london-model_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_nzlam4_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_nzlam4_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_nzlam4_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-bi-periodic_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-bi-periodic_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-bi-periodic_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-lbc_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-lbc_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-maps_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-maps_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-maps_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-non-periodic_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-non-periodic_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-non-periodic_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-op-lam_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-op-lam_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-op-lam_ex1a_cce_full-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-centres_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-centres_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-centres_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-nodes_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-nodes_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-nodes_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-points_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-points_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-stretch-points_ex1a_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-trench-x_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-trench-x_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-trench-x_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-trench-y_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_planar-trench-y_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_planar-trench-y_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_polar_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_polar_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_polar_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_uk_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_uk_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_uk_ex1a_cce_fast-debug-64bit | succeeded |
| check_mesh_tools_var-seuk_azspice_gnu_fast-debug-64bit | succeeded |
| check_mesh_tools_var-seuk_azspice_gnu_full-debug-64bit | succeeded |
| check_mesh_tools_var-seuk_ex1a_gnu_fast-debug-64bit | succeeded |
| check_simple_diffusion_default-C24_azspice_gnu_full-debug-64bit | succeeded |
| check_simple_diffusion_default-C24_ex1a_cce_full-debug-64bit | succeeded |
| check_simple_diffusion_default-C24_ex1a_gnu_full-debug-64bit | succeeded |
| check_skeleton_default-C24_azspice_gnu_full-debug-64bit | succeeded |
| check_skeleton_default-C24_ex1a_cce_full-debug-64bit | succeeded |
| check_skeleton_default-C24_ex1a_gnu_full-debug-64bit | succeeded |
| config_dump_checker | succeeded |
| export-source | succeeded |
| export-source_azspice | succeeded |
| export-source_ex1a | succeeded |
| global_variables_checker | succeeded |
| python_unit_tests | succeeded |
| remote-init_azspice | succeeded |
| remote-init_ex1a | succeeded |
| rose-stem_lint_checker | succeeded |
| run_coupled_canned_azspice_gnu_fast-debug-64bit | succeeded |
| run_coupled_canned_ex1a_cce_fast-debug-64bit | succeeded |
| run_coupled_default-C12_azspice_gnu_fast-debug-64bit | succeeded |
| run_coupled_default-C12_azspice_gnu_full-debug-64bit | succeeded |
| run_coupled_default-C12_ex1a_cce_fast-debug-64bit | succeeded |
| run_coupled_default-C12_ex1a_cce_full-debug-64bit | succeeded |
| run_coupling_unit_tests_azspice_gnu_32bit | succeeded |
| run_coupling_unit_tests_azspice_gnu_64bit | succeeded |
| run_coupling_unit_tests_ex1a_gnu_32bit | succeeded |
| run_coupling_unit_tests_ex1a_gnu_64bit | succeeded |
| run_driver_unit_tests_azspice_gnu_32bit | succeeded |
| run_driver_unit_tests_azspice_gnu_64bit | succeeded |
| run_driver_unit_tests_ex1a_gnu_32bit | succeeded |
| run_driver_unit_tests_ex1a_gnu_64bit | succeeded |
| run_infrastructure_integration_tests_azspice_gnu_32bit | succeeded |
| run_infrastructure_integration_tests_azspice_gnu_64bit | succeeded |
| run_infrastructure_integration_tests_ex1a_cce_32bit | succeeded |
| run_infrastructure_integration_tests_ex1a_cce_64bit | succeeded |
| run_infrastructure_unit_tests_azspice_gnu_32bit | succeeded |
| run_infrastructure_unit_tests_azspice_gnu_64bit | succeeded |
| run_infrastructure_unit_tests_ex1a_gnu_32bit | succeeded |
| run_infrastructure_unit_tests_ex1a_gnu_64bit | succeeded |
| run_io_demo_canned_azspice_gnu_fast-debug-64bit | succeeded |
| run_io_demo_canned_ex1a_cce_fast-debug-64bit | succeeded |
| run_io_demo_default-C24_azspice_gnu_fast-debug-32bit | succeeded |
| run_io_demo_default-C24_azspice_gnu_full-debug-64bit | succeeded |
| run_io_demo_default-C24_ex1a_cce_fast-debug-32bit | succeeded |
| run_io_demo_default-C24_ex1a_cce_full-debug-64bit | succeeded |
| run_io_demo_multifile-C24_azspice_gnu_fast-debug-32bit | succeeded |
| run_io_demo_multifile-C24_azspice_gnu_fast-debug-64bit | succeeded |
| run_io_demo_multifile-C24_ex1a_cce_fast-debug-32bit | succeeded |
| run_io_demo_multifile-C24_ex1a_cce_fast-debug-64bit | succeeded |
| run_io_demo_multifile-C24_ex1a_gnu_fast-debug-32bit | succeeded |
| run_io_demo_multifile-C24_ex1a_gnu_fast-debug-64bit | succeeded |
| run_io_demo_unit_tests_azspice_gnu_64bit | succeeded |
| run_io_demo_unit_tests_ex1a_gnu_64bit | succeeded |
| run_lbc_demo_ConstantLBC-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_ConstantLBC-lbc_azspice_gnu_full-debug-64bit | succeeded |
| run_lbc_demo_ConstantLBC-lbc_ex1a_cce_fast-debug-64bit | succeeded |
| run_lbc_demo_ConstantLBC-lbc_ex1a_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_IntegerFields-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_IntegerFields-lbc_azspice_gnu_full-debug-64bit | succeeded |
| run_lbc_demo_IntegerFields-lbc_ex1a_cce_fast-debug-64bit | succeeded |
| run_lbc_demo_IntegerFields-lbc_ex1a_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_1x1P_ex1a_cce_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_1x1P_ex1a_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_2x2P_ex1a_cce_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_2x2P_ex1a_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_8x2P_ex1a_cce_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_8x2P_ex1a_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_OutputOnLBC-lbc_azspice_gnu_full-debug-64bit | succeeded |
| run_lbc_demo_canned_azspice_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_canned_ex1a_cce_fast-debug-64bit | succeeded |
| run_lbc_demo_default-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| run_lbc_demo_default-lbc_azspice_gnu_full-debug-64bit | succeeded |
| run_lbc_demo_default-lbc_ex1a_cce_fast-debug-64bit | succeeded |
| run_lbc_demo_default-lbc_ex1a_gnu_fast-debug-64bit | succeeded |
| run_lfric_xios_integration_tests_azspice_gnu_64bit | succeeded |
| run_lfric_xios_integration_tests_ex1a_cce_64bit | succeeded |
| run_lfric_xios_unit_tests_azspice_gnu_64bit | succeeded |
| run_lfric_xios_unit_tests_ex1a_gnu_64bit | succeeded |
| run_mesh_C12_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_C12_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_C24_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_C24_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_LAM50x50-2x2_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_LAM50x50-2x2_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_lbc_1x1P_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_lbc_2x2P_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_lbc_8x2P_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_lbc_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_lbc_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_canned_cubedsphere_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_canned_planar_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c1_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c1_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c1_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c2_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c2_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c2_ex1a_cce_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c3_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c3_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-c3_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-maps_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-maps_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-maps_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-op-nonuniform_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-op-nonuniform_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-op-nonuniform_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-op_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-op_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-op_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-rotated_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-rotated_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere-rotated_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_cubedsphere_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_equator-band_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_equator-band_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_equator-band_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_equator_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_equator_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_equator_ex1a_cce_full-debug-64bit | succeeded |
| run_mesh_tools_falklands_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_falklands_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_falklands_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_lam_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_lam_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_lam_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_london-model_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_london-model_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_london-model_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_nzlam4_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_nzlam4_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_nzlam4_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-bi-periodic_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-bi-periodic_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-bi-periodic_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-lbc_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-lbc_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-lbc_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-maps_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-maps_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-maps_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-non-periodic_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-non-periodic_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-non-periodic_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-op-lam_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-op-lam_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-op-lam_ex1a_cce_full-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-centres_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-centres_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-centres_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-nodes_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-nodes_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-nodes_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-points_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-points_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-stretch-points_ex1a_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-trench-x_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-trench-x_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-trench-x_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-trench-y_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_planar-trench-y_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_planar-trench-y_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_polar_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_polar_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_polar_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_uk_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_uk_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_uk_ex1a_cce_fast-debug-64bit | succeeded |
| run_mesh_tools_unit_tests_azspice_gnu_64bit | succeeded |
| run_mesh_tools_unit_tests_ex1a_gnu_64bit | succeeded |
| run_mesh_tools_var-seuk_azspice_gnu_fast-debug-64bit | succeeded |
| run_mesh_tools_var-seuk_azspice_gnu_full-debug-64bit | succeeded |
| run_mesh_tools_var-seuk_ex1a_gnu_fast-debug-64bit | succeeded |
| run_science_unit_tests_azspice_gnu_32bit | succeeded |
| run_science_unit_tests_azspice_gnu_64bit | succeeded |
| run_science_unit_tests_ex1a_gnu_32bit | succeeded |
| run_science_unit_tests_ex1a_gnu_64bit | succeeded |
| run_simple_diffusion_canned_azspice_gnu_fast-debug-64bit | succeeded |
| run_simple_diffusion_canned_ex1a_cce_fast-debug-64bit | succeeded |
| run_simple_diffusion_default-C24_azspice_gnu_full-debug-64bit | succeeded |
| run_simple_diffusion_default-C24_ex1a_cce_full-debug-64bit | succeeded |
| run_simple_diffusion_default-C24_ex1a_gnu_full-debug-64bit | succeeded |
| run_simple_diffusion_unit_tests_azspice_gnu_64bit | succeeded |
| run_simple_diffusion_unit_tests_ex1a_gnu_64bit | succeeded |
| run_skeleton_canned_azspice_gnu_fast-debug-64bit | succeeded |
| run_skeleton_canned_ex1a_gnu_fast-debug-64bit | succeeded |
| run_skeleton_default-C24_azspice_gnu_full-debug-64bit | succeeded |
| run_skeleton_default-C24_ex1a_cce_full-debug-64bit | succeeded |
| run_skeleton_default-C24_ex1a_gnu_full-debug-64bit | succeeded |
| run_skeleton_unit_tests_azspice_gnu_64bit | succeeded |
| run_skeleton_unit_tests_ex1a_gnu_64bit | succeeded |
| site_validator | succeeded |
| style_checker | succeeded |
| validate_rose_meta | succeeded |
</details>


<!-- Paste your trac.log from testing output here -->

## Security Considerations

- [X] I have reviewed my changes for potential security issues
- [X] Sensitive data is properly handled (if applicable)
- [X] Authentication and authorisation are properly implemented (if applicable)

## Performance Impact

- [X] Performance of the code has been considered and, if applicable, suitable
      performance measurements have been conducted

## AI Assistance and Attribution

- [ ] Some of the content of this change has been produced with the assistance
      of _Generative AI tool name_ (e.g., Met Office Github Copilot Enterprise,
      Github Copilot Personal, ChatGPT GPT-4, etc) and I have followed the
      [Simulation Systems AI policy](https://metoffice.github.io/simulation-systems/FurtherDetails/ai.html)
      (including attribution labels)

<!-- If AI has been used, please provide more details here -->

## Documentation

- [ ] Where appropriate I have updated documentation related to this change and
      confirmed that it builds correctly

## PSyclone Approval

- [ ] If you have edited any PSyclone-related code (e.g. PSyKAl-lite, Kernel
      interface, optimisation scripts, LFRic data structure code) then please
      contact the
      [tooscollabdevteam@metoffice.gov.uk](tooscollabdevteam@metoffice.gov.uk)

# Sci/Tech Review

<!-- To be completed by the Sci/Tech Reviewer -->
<!-- May be skipped for trivial tickets -->

- [x] I understand this area of code and the changes being added
- [x] The proposed changes correspond to the pull request description
- [x] Documentation is sufficient (do documentation papers need updating)
- [x] Sufficient testing has been completed

_Please alert the code reviewer via a tag when you have approved the SR_

# Code Review

<!-- To be completed by the Code Reviewer -->

- [x] All dependencies have been resolved
- [x] Related Issues have been properly linked and addressed
- [x] CLA compliance has been confirmed
- [x] Code quality standards have been met
- [x] Tests are adequate and have passed
- [ ] Documentation is complete and accurate
- [ ] Security considerations have been addressed
- [ ] Performance impact is acceptable

jedbakerMO committed:
#192 : merging in fcm branch

jedbakerMO committed:
#192 : merging in fcm branch

jedbakerMO committed:
Merge remote-tracking branch 'origin/stable' into 192_timing_mod_migration_core

jedbakerMO committed:
Merge branch 'main' into 192_timing_mod_migration_core

christophermaynard reviewed (commented):
Tested and all working

jedbakerMO committed:
Reordering contributor name

Moved contributor name entry in CONTRIBUTORS.md to be alphabetical according to github username.

jedbakerMO committed:
Remove empty line in CONTRIBUTORS.md

jedbakerMO committed:
Merge branch '192_timing_mod_migration_core' of github.com:jedbakerMO/lfric_core into 192_timing_mod_migration_core

jedbakerMO committed:
Adding timing options to io_demo to ensure Vernier changes are tested

jedbakerMO committed:
Turning on timing in io_demo and fixing issue with subroutine timers off

mo-rickywong committed:
Merge branch 'main' into 192_timing_mod_migration_core

mo-rickywong committed:
Merge branch 'main' into 192_timing_mod_migration_core

mo-rickywong committed:
Merge branch 'main' into 192_timing_mod_migration_core
